### PR TITLE
DOC: Remove documentation specific to Python 2

### PR DESCRIPTION
**Pull request by DimitriPapadopoulos**
Fixes documentation issues described in #22400.

### CODE REVIEW

**Comment by mattip** [comment]
In the issue there are 5 instances mentioned, the PR removes 3 of them. Is there a reason you left the other two?

File: numpy/core/_add_newdocs.py [hunk 1]
```diff
@@ -1354,8 +1354,7 @@
             decimal numbers, an operation which is better spelt
             ``frombuffer(string, dtype, count)``. If `string` contains unicode
             text, the binary mode of `fromstring` will first encode it into
-            bytes using either utf-8 (python 3) or the default encoding
-            (python 2), neither of which produce sane results.
+            bytes using utf-8, which will not produce sane results.
```
Comment (mattip) [inline numpy/core/_add_newdocs.py:1357]: Maybe we should just remove the kwarg `sep` since it has been deprecated for long enough

**Comment by DimitriPapadopoulos** [comment]
> In the issue there are ~5~ 6 instances mentioned, the PR removes 3 of them. Is there a reason you left the others?

That's because I have been asked to fix only documentation in the merge request, at least that's what I understood:
https://github.com/numpy/numpy/issues/22400#issuecomment-1272188908

File: numpy/core/_add_newdocs.py (hunk 1 above)
Comment (DimitriPapadopoulos) [reply numpy/core/_add_newdocs.py:1357]: The purpose of this merge request was to fix documentation only, not to modify actual code. I have nothing against broadening its scope, but please make sure you all agree on the new scope.

**Comment by HaoZeke** [comment]
> > In the issue there are ~5~ 6 instances mentioned, the PR removes 3 of them. Is there a reason you left the others?
> 
> That's because I have been asked to fix only documentation in the merge request, at least that's what I understood: [#22400 (comment)](https://github.com/numpy/numpy/issues/22400#issuecomment-1272188908)

I think since the other instances are within docstrings they should be part of this PR (i.e. they should also be `DOC:` changes)

**Comment by DimitriPapadopoulos** [comment]
Some of the docstrings just inform that the following code is for Python 2. It would make sense to remove the code, not just the docstring.

**Comment by seberg** [comment]
Merging, thanks everyone.  And yes, I agree with Matti, we should just finalize that deprecation.

**Comment by DimitriPapadopoulos** [comment]
I can push a few other merge requests to remove deprecated code.

**Pull request merged**